src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -189,6 +195,8 @@ public async Task WriteStateAsync<T>(string grainType, GrainReference grainId, I
         public async Task ClearStateAsync<T>(string grainType, GrainReference grainId, IGrainState<T> grainState)
         {
             var blobName = GetBlobName(grainType, grainId);
+            var container = this.blobContainerFactory.BuildContainerClient(grainType, grainId);
```

Comment:
This seems expensive - creating a new container for each call

Reply:
Yeah I wasn't wild about this either but I can't think of another way of handling it. The default implementation wouldn't create a new container each call but I guess _could_

Reply:
Everyone wouldn't have to pay for this apart from the overhead of the method call to the factory.

Reply:
Ah you're right, my bad.